### PR TITLE
refactor(alert): add typescript

### PR DESCRIPTION
**Pull request by lionelB**
Convert alert-cli to typescript 

Next steps: 
- migrate old alerts
- use typescript in alert frontend

### CODE REVIEW

**Comment by lionelB** [comment]
```ts
to avoid the type spaghetti between agreement, code, article and section, I propose to use a new type for DilaChanges

export type DilaDataAlertChanges = { 
   type: "dila"
  modified: {
    parents: string[];
    title: string;
    id: string;
    cid: string;
    rootId: string;
    diffs: [
      { type: "nota" | "texte"; currentText: string; previousText: string }
    ];
  };
  added: {
    etat: string;
    parents: string[];
    title: string;
    id: string;
    cid: string;
    rootId: string;
  };
  removed: {
    parents: string[];
    title: string;
    id: string;
    cid: string;
    rootId: string;
  };
};
``` 
we could then migrate the existing alerts by reseting the sources versions and re-running the alert script

File: targets/alert-cli/src/diff/fiches-travail-data.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
+/* eslint-disable-next-line*/
+import { FicheTravailEmploiDoc } from "@shared/types";
```
Comment (m-maillot) [inline targets/alert-cli/src/diff/fiches-travail-data.ts:2]: Import not used.

File: targets/alert-cli/src/diff/fiches-travail-data.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
+/* eslint-disable-next-line*/
```
Comment (m-maillot) [inline targets/alert-cli/src/diff/fiches-travail-data.ts:1]: Why did you disable the import order?

File: targets/alert-cli/src/diff/fiches-vdd.ts [hunk 3]
```diff
@@ -1,69 +1,84 @@
 /* eslint-disable-next-line*/
-import nodegit from "nodegit";
+import nodegit, { ConvenientPatch, Tree } from "nodegit";
```
Comment (m-maillot) [inline targets/alert-cli/src/diff/fiches-vdd.ts:2]:
```suggestion
import { ConvenientPatch, Tree } from "nodegit";
```

**Review by m-maillot — APPROVED** [review]
Awesome work! :+1:

File: targets/alert-cli/src/diff/fiches-travail-data.ts (hunk 2 above)
Comment (lionelB) [reply targets/alert-cli/src/diff/fiches-travail-data.ts:1]: removed !

**Pull request merged**